fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricTagProvider.java

```diff
@@ -225,11 +231,21 @@ private FabricTagBuilder(ObjectBuilder<T> parent) {
 		/**
 		 * Add another tag to this tag.
 		 *
+		 * <p><b>Note:</b> any vanilla tags can be added to the builder,
+		 * but other tags can only be added if it has a builder registered in the same provider.
```

Comment:
Does datagen run before or after other mods have run through their initializer?
On Forge, other mods do run and register their stuff, making it possible to access the tags and item/block definitions of other mods on the classpath.

Reply:
Data gen should run after all mods have been initialised, if thats not the case now then something is wrong.

Reply:
It's not the case for gametests at least, not sure about datagen.

Reply:
Yes, should be the same for the gametests, might just be an oversight on my part.

Reply:
In `AbstractTagProvider#run` it checks if there's a builder for the required tag and it throws an error if not. Not sure how forge datagen works but it looks like it actually loads the resources as if it's in the world?